hudi-client/src/main/java/org/apache/hudi/HoodieWriteClient.java

```diff
@@ -325,6 +326,31 @@ public static SparkConf registerClasses(SparkConf conf) {
     }
   }
 
+  /**
+   * Deletes a bunch of keys from the Hoodie table, at the supplied commitTime
+   */
+  public JavaRDD<WriteStatus> delete(JavaRDD<HoodieKey> keys, final String commitTime) {
+    HoodieTable<T> table = getTableAndInitCtx();
+    try {
+      // De-dupe/merge if needed
+      JavaRDD<HoodieKey> dedupedKeys =
+          combineKeysOnCondition(config.shouldCombineBeforeUpsert(), keys, config.getUpsertShuffleParallelism());
+
+      JavaRDD<HoodieRecord<T>> dedupedRecords = generateHoodieRecordsToDeleteFromKeys(dedupedKeys);
+      indexTimer = metrics.getIndexCtx();
+      // perform index loop up to get existing location of records
+      JavaRDD<HoodieRecord<T>> taggedRecords = index.tagLocation(dedupedRecords, jsc, table);
```

Comment:
I think we need to do an important step here : remove record keys which does not have location. For hard-deletes, if the record-key to be deleted is not there, we can simply filter them out here

Reply:
I looked through the code and index look up automatically drops non existent records. So, I guess we are good on that.

Reply:
Can you double check. We allow inserts to happen as part of upsert() calls and it uses similar path. Also, Look at HoodieReadClient.filterExists. You need an opposite of that.

Reply:
yes, my bad. I realized that this would return all records and hence the next line where in I filer for records which has non null location.